src/main/java/com/google/sps/servlets/BusinessDataServlet.java

```diff
@@ -94,19 +96,34 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String addressLatStr = request.getParameter(BUSINESS_ADDRESS_LAT);
     String addressLngStr = request.getParameter(BUSINESS_ADDRESS_LNG);
     float addressLat;
-      if (addressLatStr.isEmpty()) {
-          addressLat = 404;
+    if (addressLatStr.isEmpty()) {
+        addressLat = 404;
+    } else {
+        addressLat = Float.parseFloat(addressLatStr);
+    }
+    float addressLng;
+    if (addressLngStr.isEmpty()) {
+        addressLng = 404;
+    } else {
+        addressLng = Float.parseFloat(addressLngStr);
+    }
+    String contactDetails = request.getParameter(BUSINESS_CONTACT_INFO);
+    String orderDetails = request.getParameter(BUSINESS_ORDER_INFO);
+    String minPriceStr = request.getParameter(BUSINESS_MIN_PRICE);
+    String maxPriceStr = request.getParameter(BUSINESS_MAX_PRICE);
+    float minPrice;
+      if (minPriceStr.isEmpty()) {
+          minPrice = 404;
       } else {
-          addressLat = Float.parseFloat(addressLatStr);
+          minPrice = Float.parseFloat(minPriceStr);
       }
-      float addressLng;
-      if (addressLngStr.isEmpty()) {
-          addressLng = 404;
+      float maxPrice;
+      if (maxPriceStr.isEmpty()) {
```

Comment:
If the default value is all 404, can consider having a function which returns 404 when empty so its neater.

Reply:
Sounds good! I'll implement this and submit a pull request

Reply:
i mean merge this in